src/safeds/data/tabular/containers/_time_series.py

```diff
@@ -0,0 +1,848 @@
+    @staticmethod
+    def _from_tagged_table(
+        tagged_table: TaggedTable,
+        time_name: str,
+    ) -> TimeSeries:
+        """Create a time series from a tagged table.
+
+        Parameters
+        ----------
+        table : TaggedTable
+            The tagged table.
+        time_name: str
+            Name of the time column.
+
+        Returns
+        -------
+        time_series : TimeSeries
+            the created time series
+
+        Raises
+        ------
+        UnknownColumnNameError
+            If time_name matches none of the column names.
+        Value Error
+            If time column is also a feature column
+
+        Examples
+        --------
+        >>> from safeds.data.tabular.containers import Table, TimeSeries
+        >>> tagged_table = TaggedTable({"date": ["01.01", "01.02", "01.03", "01.04"], "col1": ["a", "b", "c", "a"]}, "col1" )
+        >>> timeseries = TimeSeries._from_tagged_table(tagged_table, time_name = "date")
+        """
+        if time_name not in tagged_table.column_names:
+            raise UnknownColumnNameError([time_name])
+        table = tagged_table._as_table()
+        # make sure that the time_name is not part of the features
+        result = object.__new__(TimeSeries)
+        feature_names = tagged_table.features.column_names
+        if time_name in feature_names:
+            feature_names.remove(time_name)
+
+        if time_name == tagged_table.target.name:
+            raise ValueError(f"Column '{time_name}' cannot be both time column and target.")
+
+        result._data = table._data
+        result._schema = table.schema
+        result._time = table.get_column(time_name)
+        result._features = table.keep_only_columns(feature_names)
+        result._target = table.get_column(tagged_table.target.name)
+        return result
+
+    @staticmethod
+    def _from_table_to_time_series(
+        table: Table,
+        target_name: str,
+        time_name: str,
+        feature_names: list[str] | None = None,
+    ) -> TimeSeries:
+        """Create a TimeSeries from a table.
+
+        Parameters
+        ----------
+        table : Table
+            The table.
+        target_name : str
+            Name of the target column.
+        time_name: str
+            Name of the date column.
+        feature_names : list[str] | None
+            Names of the feature columns. If None, all columns except the target and time columns are used.
+
+        Returns
+        -------
+        time_series : TimeSeries
+            the created time series
+
+        Raises
+        ------
+        UnknownColumnNameError
+            If target_name or time_name matches none of the column names.
+        Value Error
+            If there is no other column than the specified target and time columns left to be a feature column
+        Value Error
+            If one column is target and feature
+        Value Error
+            If time column is also feature column
```

Comment:
```suggestion
        Value Error
            If one column is target and feature
        Value Error
            If one column is time and feature
```
Choose a consistent phrase